code/modules/mob/living/carbon/xenomorph/castes/slime/abilities_slime.dm

```diff
@@ -0,0 +1,480 @@
+	SIGNAL_HANDLER
+	UnregisterSignal(owner, list(COMSIG_MOVABLE_IMPACT, COMSIG_MOVABLE_POST_THROW))
+
+/// Applies effects over a set amount of time.
+/// Will call itself again based on SLIME_POUNCE_INFLICTION_RATE unless there is no longer an attached atom.
+/datum/action/xeno_action/activable/slime_pounce/proc/effect_over_time(stacks)
+	if(!attached_atom || !owner.buckled)
+		return
+	addtimer(CALLBACK(src, PROC_REF(effect_over_time), stacks), SLIME_POUNCE_INFLICTION_RATE)
+	var/mob/living/attached_living = attached_atom
+	if(attached_living.has_status_effect(STATUS_EFFECT_INTOXICATED))
+		var/datum/status_effect/stacking/intoxicated/debuff = attached_living.has_status_effect(STATUS_EFFECT_INTOXICATED)
+		debuff.add_stacks(stacks)
+		return
+	attached_living.apply_status_effect(STATUS_EFFECT_INTOXICATED, stacks)
+
+/// Calls attached_atom_ex_act via INVOKE_ASYNC.
+/datum/action/xeno_action/activable/slime_pounce/proc/call_attached_atom_ex_act(severity, epicenter_dist, impact_range)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(attached_atom_ex_act), severity, epicenter_dist, impact_range)
+
+/// Called whenever the Slime or the atom it's attached to is hit by an explosion.
+/// Unbuckles the Slime and reflects the incoming damage.
+/datum/action/xeno_action/activable/slime_pounce/proc/attached_atom_ex_act(severity, epicenter_dist, impact_range)
+	end_ability()
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			xeno_owner.apply_damage(INFINITY, BRUTE, blocked = BOMB)
+		if(EXPLODE_HEAVY)
+			xeno_owner.apply_damage(rand(100, 250), BRUTE, blocked = BOMB)
+		if(EXPLODE_LIGHT)
+			xeno_owner.apply_damage(rand(10, 90), BRUTE, blocked = BOMB)
+
+/// Calls attached_atom_projectile_hit via INVOKE_ASYNC.
+/datum/action/xeno_action/activable/slime_pounce/proc/call_attached_atom_projectile_hit(datum/source, obj/projectile/proj)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(attached_atom_projectile_hit), source, proj)
+
+/// Called whenever the atom the Slime is attached to is hit by a projectile.
+/// Unbuckles if the projectile is fire, and reflects the incoming damage.
+/datum/action/xeno_action/activable/slime_pounce/proc/attached_atom_projectile_hit(datum/source, obj/projectile/proj)
+	if(proj.ammo.flags_ammo_behavior & AMMO_EXPLOSIVE || proj.ammo.flags_ammo_behavior & AMMO_ROCKET || proj.ammo.flags_ammo_behavior & AMMO_INCENDIARY || proj.ammo.flags_ammo_behavior & AMMO_FLAME) // If hit by an explosive, automatically unbuckle.
+		end_ability()
+		return
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	var/blocked_type = determine_armor_type(proj) // The corresponding armor type for the projectile.
+	xeno_owner.apply_damage(proj.damage, proj.ammo.damage_type, blocked = blocked_type, penetration = proj.ammo.penetration)
+	animation_flash_color(xeno_owner)
+	if(proj.ammo.sound_hit)
+		playsound(xeno_owner, proj.ammo.sound_hit, 50, 1)
```

Comment:
Dont you want to just call the projectile_hit proc or whatever on the owner here?

Reply:
Checked this out. `projectile_hit` returns a probability of getting hit, which we don't need to check for since the signal doesn't get sent unless the attached atom to is hit. Did you mean some other proc?

Reply:
Yeah, `bullet_act` seems to be it. Donezo.